src/lib.rs

```diff
@@ -56,23 +71,39 @@ impl bb8::ManageConnection for MemcacheConnectionManager {
     type Error = io::Error;
 
     async fn connect(&self) -> Result<Self::Connection, Self::Error> {
-        Connection::connect(&self.uri).await
+        match (self.read_timeout, self.write_timeout) {
+            (Some(read_timeout), Some(write_timeout)) => {
+                match Connection::connect(&self.uri).await {
+                    Ok(connection) => Ok(connection.with_timeouts(read_timeout, write_timeout)),
+                    Err(e) => return Err(e),
+                }
+            }
+            _ => Connection::connect(&self.uri).await,
```

Comment:
ditto above, it's better to change `connect` so we can just call `Connection::connect(&self.uri, self.read_timeout, self.write_timeout)`

Reply:
yeah, looks less clunky. thanks